packages/grafana-data/src/types/datasource.ts

```diff
@@ -322,7 +330,11 @@ abstract class DataSourceApi<
 
   /** Get an identifier object for this datasource instance */
   getRef(): DataSourceRef {
-    return { type: this.type, uid: this.uid };
+    const ref: DataSourceRef = { type: this.type, uid: this.uid };
```

Comment:
nit: since `DataSourceRef.apiversion` is optional you should feel free to simply return

`return { type: this.type, uid: this.uid, apiVersion: this.apiversion }`